### PR TITLE
Timeouts configured by annotation

### PR DESCRIPTION
**Pull request by apolit**

### CODE REVIEW

**Comment by tmszdmsk** [comment]
I don't get setting a timeout on the provider side. 
It's a clients responsibility to figure out and configure the timeouts for services they use in the clients pattern of usage and requirements.

**Comment by apolit** [comment]
The timeout settings are applied at client side on proxy generation. You may got a wrong impression because in test there is only one interface shared between client and server side.

**Comment by tmszdmsk** [comment]
but as a convention api modules are created as an implementation sibling and clients import them instead of recreating interfaces and dto's on their own. It means that timeouts will be applied to all of the clients. I would rather set @Timeout at the injected field. It's not an ideal solution (because timeout will be applied to all of the methods on interface) but it's better than this.

File: tests/src/test/java/com/payu/ratel/client/standalone/tests/StandaloneClientTest.java [hunk 1]
```diff
@@ -98,7 +98,7 @@ public void shouldUseAddedServiceCallListenerToInjectedStandaloneClientFactory()
 
         // given
         // client side
-        TestService testService = standaloneFactory.getServiceProxy(TestService.class);
+        TestService testService = standaloneFactory.getServiceProxy(TestService.class, null);
```
Comment (slawekjaranowski) [inline tests/src/test/java/com/payu/ratel/client/standalone/tests/StandaloneClientTest.java]: We should test both case old and new way.

File: tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java [hunk 2]
```diff
@@ -119,7 +119,7 @@ public void shouldCreateStandaloneClientWithZookeeper() {
         RatelClientFactory ratelStandaloneFactory = RatelStandaloneFactory.fromZookeeperServer(ZK_HOST);
 
         //when
-        TestService testedService = ratelStandaloneFactory.getServiceProxy(TestService.class);
+        TestService testedService = ratelStandaloneFactory.getServiceProxy(TestService.class, null);
```
Comment (slawekjaranowski) [inline tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java]: Maybe without null will be look better.

File: tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java [hunk 3]
```diff
@@ -130,7 +130,9 @@ public void shouldResolveZookeeperHostFromCliProperty() {
         System.setProperty(SERVICE_DISCOVERY_ZK_HOST, ZK_HOST);
 
         //when
-        final TestService testedService = RatelStandaloneFactory.fromZookeeperServer().getServiceProxy(TestService.class);
+        final TestService testedService = RatelStandaloneFactory.fromZookeeperServer().getServiceProxy(
+                TestService.class,
+                null);
```
Comment (slawekjaranowski) [inline tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java]: Maybe without null will be look better.

File: tests/src/test/java/com/payu/ratel/client/standalone/tests/StandaloneClientTest.java (hunk 1 above)
Comment (apolit) [inline tests/src/test/java/com/payu/ratel/client/standalone/tests/StandaloneClientTest.java]: Tests are in com.payu.ratel.tests.timeout package

File: tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java (hunk 2 above)
Comment (apolit) [inline tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java]: Done

File: tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java (hunk 3 above)
Comment (apolit) [inline tests/src/test/java/com/payu/ratel/tests/zookeeper/ZookeeperServicePublishingTest.java]: Done

**Pull request merged**